src/test/ui/lint-anon-param-edition.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// tests that the anonymous_parameters lint is warn-by-default on the 2018 edition
+
+// compile-pass
+// compile-flags: --edition=2018
```

Comment:
One last thing, @mark-i-m! We now have `rustfix` test support. If you add `// run-rustfix` here, it will actually try applying your suggestion. The result will appear in a `lint-anon-param-edition.fixed` file (you might need to run `./x.py test` with the `--bless` option to auto-create that file though).

Reply:
Can you paste the rustc output of running your branch on this file? Preferably also the json output.

Reply:
```
warning: anonymous parameters are deprecated and will be removed in the next edition.
  --> /nobackup/rust2/src/test/ui/lint-anon-param-edition.rs:18:11
   |
LL |     fn foo(u8);
   |           ^ help: Try naming the parameter or explicitly ignoring it: `_: u8`
   |
   = note: #[warn(anonymous_parameters)] on by default
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #41686 <https://github.com/rust-lang/rust/issues/41686>
```

```
{
  "message": "anonymous parameters are deprecated and will be removed in the next edition.",
  "code": {
    "code": "anonymous_parameters",
    "explanation": null
  },
  "level": "warning",
  "spans": [
    {
      "file_name": "\/nobackup\/rust2\/src\/test\/ui\/lint-anon-param-edition.rs",
      "byte_start": 638,
      "byte_end": 639,
      "line_start": 18,
      "line_end": 18,
      "column_start": 11,
      "column_end": 12,
      "is_primary": true,
      "text": [
        {
          "text": "    fn foo(u8);",
          "highlight_start": 11,
          "highlight_end": 12
        }
      ],
      "label": null,
      "suggested_replacement": null,
      "expansion": null
    }
  ],
  "children": [
    {
      "message": "#[warn(anonymous_parameters)] on by default",
      "code": null,
      "level": "note",
      "spans": [
        
      ],
      "children": [
        
      ],
      "rendered": null
    },
    {
      "message": "this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!",
      "code": null,
      "level": "warning",
      "spans": [
        
      ],
      "children": [
        
      ],
      "rendered": null
    },
    {
      "message": "for more information, see issue #41686 <https:\/\/github.com\/rust-lang\/rust\/issues\/41686>",
      "code": null,
      "level": "note",
      "spans": [
        
      ],
      "children": [
        
      ],
      "rendered": null
    },
    {
      "message": "Try naming the parameter or explicitly ignoring it",
      "code": null,
      "level": "help",
      "spans": [
        {
          "file_name": "\/nobackup\/rust2\/src\/test\/ui\/lint-anon-param-edition.rs",
          "byte_start": 638,
          "byte_end": 639,
          "line_start": 18,
          "line_end": 18,
          "column_start": 11,
          "column_end": 12,
          "is_primary": true,
          "text": [
            {
              "text": "    fn foo(u8);",
              "highlight_start": 11,
              "highlight_end": 12
            }
          ],
          "label": null,
          "suggested_replacement": "_: u8",
          "expansion": null
        }
      ],
      "children": [
        
      ],
      "rendered": null
    }
  ],
  "rendered": "warning: anonymous parameters are deprecated and will be removed in the next edition.\n  --> \/nobackup\/rust2\/src\/test\/ui\/lint-anon-param-edition.rs:18:11\n   |\nLL |     fn foo(u8);\n   |           ^ help: Try naming the parameter or explicitly ignoring it: `_: u8`\n   |\n   = note: #[warn(anonymous_parameters)] on by default\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #41686 <https:\/\/github.com\/rust-lang\/rust\/issues\/41686>\n\n"
}
```

Reply:
Looks like a parsing bug with the span? I.e. not your fault.

Reply:
@Manishearth I was wondering if this was because I used `arg.pat.span` as a the span for the suggestion. Since there is no `pat` (the arg is anonymous), maybe it just returns the span of the `(`? Should I use `arg.ty.span` instead?